Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -11,46 +11,44 @@ enum JavascriptError: Error {
     case invalid
 }
 
-extension WKUserScript {
-    public class func createInDefaultContentWorld(source: String, injectionTime: WKUserScriptInjectionTime, forMainFrameOnly: Bool) -> WKUserScript {
-        if #available(iOS 14.0, *) {
-            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly, in: .defaultClient)
-        } else {
-            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly)
-        }
-    }
-}
-
 public extension WKWebView {
-    func generateJavascriptFunctionString(functionName: String, args: [Any], escapeArgs: Bool = true) -> (javascript: String, error: Error?) {
-        let context = JSContext()
-
-        var sanitizedArgs: [String] = []
-        var error: Error?
-
-        args.forEach {
-            if !escapeArgs {
-                sanitizedArgs.append("\($0)")
-                return
-            }
-            context?.exceptionHandler = { context, exception in
-                if exception != nil {
-                    error = JavascriptError.invalid
+    func generateJSFunctionString(functionName: String, args: [Any?], escapeArgs: Bool = true) -> (javascript: String, error: Error?) {
+        var sanitizedArgs = [String]()
+        for arg in args {
+            if let arg = arg {
+                do {
+                    if let arg = arg as? String {
+                        sanitizedArgs.append(escapeArgs ? "'\(arg.htmlEntityEncodedString)'" : "\(arg)")
```

Comment:
Isn't it enough to use the JSONDecoder instead of the custom htmlEntityEncodedString?

https://stackoverflow.com/questions/59236554/how-to-use-swifts-jsonserialization-when-json-returns-a-single-string

Reply:
The original code was using the `htmlEntityEncodedString` I believe by itself, the `JSONDecoder` or `JSONEncoder` doesn't escape `'` character. Only `"`.